File: GeneratedEntityReaderClasses/Components/InheritTransformComponent.cs
namespace GeneratedNoitaClasses;
public class InheritTransformComponent : NoitaComponentBase,  iNoitaType<InheritTransformComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaBool use_root_parent { get; set; } = new();
	public NoitaBool only_position { get; set; } = new();
	public NoitaString parent_hotspot_tag { get; set; } = new();
	public NoitaInt parent_sprite_id { get; set; } = new();
	public NoitaBool always_use_immediate_parent_rotation { get; set; } = new();
	public NoitaBool rotate_based_on_x_scale { get; set; } = new();
	public NoitaCXForm<NoitaFloat> Transform { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		use_root_parent.Read(s);
		only_position.Read(s);
		parent_hotspot_tag.Read(s);
		parent_sprite_id.Read(s);
		always_use_immediate_parent_rotation.Read(s);
		rotate_based_on_x_scale.Read(s);
		Transform.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		use_root_parent.Write(s);
		only_position.Write(s);
		parent_hotspot_tag.Write(s);
		parent_sprite_id.Write(s);
		always_use_immediate_parent_rotation.Write(s);
		rotate_based_on_x_scale.Write(s);
		Transform.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/InheritTransformComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaCXForm
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class InheritTransformComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    use_root_parent:NoitaBool = field(default_factory=NoitaBool)
    only_position:NoitaBool = field(default_factory=NoitaBool)
    parent_hotspot_tag:NoitaString = field(default_factory=NoitaString)
    parent_sprite_id:NoitaInt = field(default_factory=NoitaInt)
    always_use_immediate_parent_rotation:NoitaBool = field(default_factory=NoitaBool)
    rotate_based_on_x_scale:NoitaBool = field(default_factory=NoitaBool)
    Transform:NoitaCXForm = field(default_factory=lambda: NoitaCXForm(NoitaFloat))
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.use_root_parent.Read(s)
        self.only_position.Read(s)
        self.parent_hotspot_tag.Read(s)
        self.parent_sprite_id.Read(s)
        self.always_use_immediate_parent_rotation.Read(s)
        self.rotate_based_on_x_scale.Read(s)
        self.Transform.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.use_root_parent.Write(s)
        self.only_position.Write(s)
        self.parent_hotspot_tag.Write(s)
        self.parent_sprite_id.Write(s)
        self.always_use_immediate_parent_rotation.Write(s)
        self.rotate_based_on_x_scale.Write(s)
        self.Transform.Write(s)

